core/corehttp/gateway_handler.go

```diff
@@ -152,12 +152,6 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 		ipnsHostname = true
 	}
 
-	if i.config.BlockList != nil && i.config.BlockList.ShouldBlock(urlPath) {
-		w.WriteHeader(http.StatusForbidden)
-		w.Write([]byte("403 - Forbidden"))
-		return
-	}
```

Comment:
There are also proper comments around this very important piece of code now (see above) and for the tests too: https://github.com/ipfs/go-ipfs/blob/master/test/sharness/t0400-api-security.sh